src/App.tsx

```diff
@@ -1,24 +1,77 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  clockName: number;
+  visibility: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    clockName: 0,
+    visibility: true,
+  };
+
+  showClock = () => {
+    this.setState({
+      visibility: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      visibility: false,
+    });
+  };
+
+  changeName = () => {
+    this.setState({ clockName: Math.floor(Math.random() * 100) });
+  };
+
+  render() {
+    const { clockName, visibility } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__container">
+          <h1 className="App__title">React clock</h1>
+          <p className="App_clock">
+            Random name:
+            {' '}
+            {visibility && (
+              <Clock clockName={clockName} />
+            )}
+          </p>
```

Comment:
Move that `<p>` inside Clock component, cause what I see when clock is hidden is just `Random name:`